scipy/sparse/csgraph/_matching.pyx

```diff
@@ -0,0 +1,276 @@
+cimport cython
+import numpy as np
+cimport numpy as np
+
+
+from scipy.sparse import (csr_matrix,
+                          isspmatrix_coo, isspmatrix_csc, isspmatrix_csr)
+
+include "parameters.pxi"
+
+
+def maximum_bipartite_matching(graph, perm_type='row'):
+    r"""
+    maximum_bipartite_matching(graph, perm_type='row')
+
+    Returns a matching of a bipartite graph whose cardinality is as least that
+    of any given matching of the graph.
+
+    Parameters
+    ----------
+    graph : sparse matrix
+        Input sparse in CSR format whose rows represent one partition of the
+        graph and whose columns represent the other partition. An edge between
+        two vertices is indicated by the corresponding entry in the matrix
+        existing in its sparse representation.
+    perm_type : str, {'row', 'column'}
+        Which partition to return the matching in terms of: If ``'row'``, the
+        function produces an array whose length is the number of columns in the
+        input, and whose :math:`j`'th element is the row matched to the
+        :math:`j`'th column. Conversely, if ``perm_type`` is ``'column'``, this
+        returns the columns matched to each row.
+
+    Returns
+    -------
+    perm : ndarray
+        A matching of the vertices in one of the two partitions. Unmatched
+        vertices are represented by a ``-1`` in the result.
+
+    Notes
+    -----
+    This function uses of the Hopcroft--Karp algorithm [1]_. Its time
+    complexity is :math:`O(\lvert E \rvert \sqrt{\lvert V \rvert})`, and its
+    space complexity is linear in the number of rows. In practice, this
+    asymmetry between rows and columns means that it can be more efficient to
+    transpose the input if it contains more columns than rows.
+
+    By Konig's theorem, the cardinality of the matching is also the number of
+    vertices appearing in a minimum vertex cover of the graph.
+
+    Note that if the sparse representation contains explicit zeros, these are
+    still counted as edges.
+
+    .. versionadded:: 0.15.0
```

Comment:
this can be removed by now. I suggest to replace with a note that the behavior became more general in 1.4.0. That way users are aware that if they write code against 1.4.0 it won't necessarily work on older versions